tv_apps/fling-player/test/unit/connector_test.js

```diff
@@ -0,0 +1,335 @@
+    });
+  });
+
+  suite('reportStatus()', function () {
+
+    var prevSeq, currSeq;
+
+    function assertReportStatus(spy, msg, prevSeq, currSeq) {
+      assert.isTrue(
+        spy.withArgs(msg).calledOnce,
+        'Not called once with specified msg arg'
+      );
+      assert.equal(
+        prevSeq + 1, currSeq,
+        'Seq of status msg does not increase after reporting'
+      );
+    }
+
+    setup(function () {
+      spy = sinon.spy(connector, 'sendMsg');
+      msg = castingMsgTemplate.get().statusBuffering;
+      prevSeq = connector.reportStatus(msg.status, { time : msg.time });
+    });
+
+    test('should report status of buffering', function () {
+      msg = castingMsgTemplate.get().statusBuffering;
+      currSeq = connector.reportStatus(msg.status, { time : msg.time });
+      assertReportStatus(spy, msg, prevSeq, currSeq);
+    });
+
+    test('should report status of loaded', function () {
+      msg = castingMsgTemplate.get().statusLoaded;
+      currSeq = connector.reportStatus(msg.status, { time : msg.time });
+      assertReportStatus(spy, msg, prevSeq, currSeq);
+    });
+
+    test('should report status of buffered', function () {
+      msg = castingMsgTemplate.get().statusBuffered;
+      currSeq = connector.reportStatus(msg.status, { time : msg.time });
+      assertReportStatus(spy, msg, prevSeq, currSeq);
+    });
+
+    test('should report status of playing', function () {
+      msg = castingMsgTemplate.get().statusPlaying;
+      currSeq = connector.reportStatus(msg.status, { time : msg.time });
+      assertReportStatus(spy, msg, prevSeq, currSeq);
+    });
+
+    test('should report status of seeked', function () {
+      msg = castingMsgTemplate.get().statusSeeked;
+      currSeq = connector.reportStatus(msg.status, { time : msg.time });
+      assertReportStatus(spy, msg, prevSeq, currSeq);
+    });
+
+    test('should report status of stopped', function () {
+      msg = castingMsgTemplate.get().statusStopped;
+      currSeq = connector.reportStatus(msg.status, { time : msg.time });
+      assertReportStatus(spy, msg, prevSeq, currSeq);
+    });
+
+    test('should report status of error', function () {
+      msg = castingMsgTemplate.get().statusError;
+      currSeq = connector.reportStatus(msg.status, {
+        time : msg.time,
+        error : msg.error
+      });
+      assertReportStatus(spy, msg, prevSeq, currSeq);
+    });
+
+    test('should not report status of unknown', function () {
+      msg = castingMsgTemplate.get().statusPlaying;
+      msg.status = 'unknown';
+      expect(() => connector.reportStatus(msg.status, { time : msg.time }))
+            .to.throw(/Ilegal status/);
+    });
+
+    test('should not report status without time', function () {
+      msg = castingMsgTemplate.get().statusPlaying;
+      expect(() => connector.reportStatus(msg.status, {}))
+            .to.throw(/Ilegal time/);
+    });
+  });
+
+  suite('handleRemoteMessage()', function () {
```

Comment:
This is actually testing `onConnectionMessage` altogether with `handleRemoteMessage`. I would suggest to split them up if possible.